docs/design/w3c/spec.md

```diff
@@ -0,0 +1,530 @@
+that result in machine-readable [JSON-LD](https://www.w3.org/TR/vc-data-model/#json-ld) information about the object
+format.
+
+The **context** definition used for AnonCreds W3C credentials representation can be found [here](./context.json).
+
+In the case of W3C AnonCreds credentials, the `@context` attribute includes an extra
+entry `https://github.io/anoncreds-w3c/context.json`
+which is required for the resolution of custom structure definitions and looks the following:
+
+```
+{
+  ...  
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://github.io/anoncreds-w3c/context.json"
+  ],
+  ...
+}
+```
+
+#### Identifiers
+
+W3C [Identifiers](https://www.w3.org/TR/vc-data-model/#identifiers) section defines an optional capability to assign
+some kind of identifier to the verifiable credential so that others can express statements about the same thing.
+
+In the case of W3C AnonCreds credentials, the `id` attribute is not recommended to set.
+
+#### Types
+
+W3C [Types](https://www.w3.org/TR/vc-data-model/#types) section requires including of `type` property to verifiable
+credential.
+The value of the `type` property must be one or more [URI](https://www.w3.org/TR/vc-data-model/#dfn-uri) resolvable
+through the defined `@context` to the information required for determining whether a verifiable credential or verifiable
+presentation has a valid structure.
+
+In the case of W3C AnonCreds credentials, the `type` attribute includes an extra entry `AnonCredsCredential`
+pointing to the difference in a base credential structure and looks the following:
+
+```
+{
+  ... 
+  "type": [
+    "VerifiableCredential",         // general verifiable credential definition
+    "AnonCredsCredential",          // definition for AnonCreds credentials
+  ]
+  ...
+}
+```
+
+#### Credential Subject
+
+W3C [Credential Subject](https://www.w3.org/TR/vc-data-model/#credential-subject) section requires including
+of `credentialSubject` property to verifiable credential.
+
+Credential subject value contains [claims](https://www.w3.org/TR/vc-data-model/#claims) about one or more subjects.
+
+In the context of W3C Announced credentials, credential subject property is compliant with the following statements:
+
+- credentials will always include claims about only one subjects.
+    - So that `credentialSubject` property will always be represented as an object entry, but not an array.
+- credentials claims are always represented as key value pairs, where `value` is the `raw` value of CL credential
+  attributes.
+    - encoded CL credential values are not included in the credential subject
+
+In the case of W3C AnonCreds credentials, the `credentialSubject` attribute looks the following:
+
+```
+{
+  ... 
+  "credentialSubject": {
+    "name": "Alice Jones",
+  }
+  ...
+}
+```
+
+* **TO DISCUSS**: W3C data model allows attributes to be represented not only as key/value string pairs but also as
```

Comment:
Per the goal of this effort, any AnonCreds v1 credential can be represented as an AnonCreds in W3C VC format, but not the reverse — an arbitrary W3C VC format credential (notably, those with nested attributes and arrays) will not be supported in this version.

Longer term — beyond this immediate work — arrays are out, as they are not supported by CL Signatures, and they are extremely difficult to request/present.  Nested attributes could easily be supported using a dot-notation attribute list.

As always, those for whom array support is necessary can do so with an “rich attribute value” — e.g. use a data url to encode JSON into the array.  Can’t selectively disclose the array items, but data could be transported in the VC/VP.

Reply:
> arrays are out, as they are not supported by CL Signatures

I'm curious. What is the reason for using a different signing key for every attribute in an anoncreds credential? Because if the same signing key would be used for all array entries under a key it should be quite doable to supports arrays right?